### ISSUE TITLE
PLC1901 false-positives with pandas

### ISSUE THREAD

**Issue by twoertwein** [charliermarsh/ruff#3494]
```py
import pandas as pd

data =  pd.DataFrame({"a": ["", "abc", "xyz"]})

data.loc[data["a"] != ""]  # PLC1901
# data.loc[not data["a"]]   # ValueError at runtime
```
An option to avoid this false-positive would be to ignore `!= ""` comparisons when they are inside `[ ]`.

edit: or nudge people to `data.loc[data["a"].ne("")] `
